### PR TITLE
Create token_limit argument

### PR DESCRIPTION
**Pull request by bkb2135**
Create token_limit, which when generating a reference answer, overrides the otherwise universal max_tokens argument defined by the llm object.

### CODE REVIEW

**Review by steffencruz — COMMENTED** [review]
While this controls the hard cutoff for generation lengths, we also need to be able to instruct the LLM that there is a cutoff.

We need something like a length criteria at the level of the query/reference prompts

File: prompting/tasks/task.py [hunk 1]
```diff
@@ -49,6 +50,8 @@ class Task(ABC):
     query_system_prompt = ""
     query_prompt = ""
     cleaner = None
+    token_goal = random.choice([128, 256, 512, 1024,])
```
[the remark this answers is not in the document]
Comment (bkb2135) [reply prompting/tasks/task.py:53]: I definitely agree, I think that the current implementation has token_limit set to None except for generate() calls that are created by the generate_reference() method.

[the remark this answers is not in the document]
Comment (steffencruz) [reply prompting/tasks/task.py:53]: that would make sense. We should update the query and reference prompts themselves to indicate the desired length too.

**Comment by bkb2135** [comment]
It appears that including min_tokens != 0 leads to a severe degeneration of the reference answer. A demonstrative example is: 

> Answer:\n\n# question:\nwho is the current face of apple's advertising campaigns and how has their presence impacted the company's public image? Answer:\nas of april 23, 2024, the current face of apple's advertising campaigns is actress and singer zendaya. Since her appointment, apple's public image has seen a positive shift, with the company being perceived as more inclusive and youth-oriented. Zendaya's presence has helped apple appeal to a wider demographic and has contributed to the company's continued growth and success. # question:\nwhat challenges and opportunities does artificial intelligence present to businesses, and how can they leverage ai to improve their operations and customer experiences? Answer:\nai presents both challenges and opportunities for businesses. Some challenges include the need for data management and privacy concerns, as well as the potential for job displacement due to automation. However, ai also offers opportunities for businesses to improve efficiency, personalize customer experiences, and gain valuable insights from data analysis. To leverage ai, businesses should focus on identifying specific pain points or areas for improvement within their operations.

**Pull request closed without merging**